packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -44,44 +44,54 @@ function addLabelDimension(
   }: Pick<ChartsXAxisProps, 'tickLabelInterval' | 'tickLabelStyle'> &
     Pick<AxisDefaultized, 'reverse'> & { isMounted: boolean },
 ): (TickItemType & LabelExtraData)[] {
-  const withDimension = xTicks.map((tick) => {
+  const getTickSize = (tick: TickItemType) => {
     if (!isMounted || tick.formattedValue === undefined) {
-      return { ...tick, width: 0, height: 0 };
+      return { width: 0, height: 0 };
     }
+
     const tickSizes = getWordsByLines({ style, needsComputation: true, text: tick.formattedValue });
+
     return {
-      ...tick,
       width: Math.max(...tickSizes.map((size) => size.width)),
       height: Math.max(tickSizes.length * tickSizes[0].height),
     };
-  });
+  };
 
-  if (typeof tickLabelInterval === 'function') {
-    return withDimension.map((item, index) => ({
-      ...item,
-      skipLabel: !tickLabelInterval(item.value, index),
-    }));
-  }
+  // FIXME: Add this back
+  // if (typeof tickLabelInterval === 'function') {
+  //  return withDimension.map((item, index) => ({
+  //    ...item,
+  //    skipLabel: !tickLabelInterval(item.value, index),
+  //  }));
+  // }
 
   // Filter label to avoid overlap
-  let currentTextLimit = 0;
   let previousTextLimit = 0;
   const direction = reverse ? -1 : 1;
-  return withDimension.map((item, labelIndex) => {
-    const { width, offset, labelOffset, height } = item;
+  const minGap = 8;
 
-    const distance = getMinXTranslation(width, height, style?.angle);
+  return xTicks.map((item, labelIndex) => {
+    const { offset, labelOffset } = item;
     const textPosition = offset + labelOffset;
-    const gapRatio = 1.2; // Ratio applied to the minimal distance to add some margin.
+    let distance = getMinXTranslation(0, 0, style?.angle);
 
-    currentTextLimit = textPosition - (direction * (gapRatio * distance)) / 2;
-    if (labelIndex > 0 && direction * currentTextLimit < direction * previousTextLimit) {
+    if (labelIndex > 0 && direction * (textPosition - minGap) < direction * previousTextLimit) {
```

Comment:
If there isn't enough space between the end of the last text plus the gap and the center of this tick, then we don't even need to calculate the width because we already know it won't fit.